### ISSUE TITLE
Dedicated Server: Some blocks randomly turn into other blocks or blocks disappear off structures

### ISSUE THREAD

**Issue by RavienGaming** [RavienGaming/Reforged-Eden-2#10]
### Reforged Eden 2 Build Number

1

### Game mode

Survival, Creative, Dedicated Server

### Tested in vanilla

Not tested in vanilla or not applicable

### Description of issue

This is an issue that sometimes but not always randomly and inconsistently happens after a game update or a scenario update.
Typically involves cruise missile turrets changing into hydroponics bays, but can affect other blocks as well.
There is no known cause or preventative fix at this time.
Try restarting your server and then manually fixing issues as they come up by replacing blocks or ships for players who encounter this issue.

**Update 3/9/2026:**
If you have any information that can help track down the cause or how to prevent this issue from happening, please leave a reply here. It would be most appreciated.
Due to the bug report being marked as not a bug, we can only assume it's due to either a problem with server setup or management, or a problem in the scenario itself. At this time I have no idea what is causing it or how to prevent it from happening in the future so I'm asking the community for help. Please reply to this issue if you've found a way to reproduce and/or prevent this issue from occurring, or if you have any new information that can shed some light on why this happens.

Link to relevant bug report: https://empyriononline.com/threads/blocks-randomly-turning-into-other-blocks-mp.103070/

Video where this is seen happening in real time: https://youtu.be/doYz7VjTPjE?si=yLGjkPRiPqjrKmPt

**Update 3/29/2026:**
Marking this "fixed" because while the underlying issue is most definitely not fixed, going forward we will only add new blocks to the very end of the block config file which means for new save games when the issue does occur block IDs won't change.
This unfortunately won't help any save games started before the current build 44 of RE2, but any saves started in build 44 or later should not suffer block ID changes when the block mapping file can't be read.

Also keeping the "game bug" label until or unless the game devs fix the actual problem, but that is unlikely.

I will keep this issue open for a few months to give existing servers a chance to finish up their current seasons, in case this issue happens again for them so it's easier to find this issue.

**Comment by RavienGaming** [comment]
Thank you for your report!
This issue is likely a game bug and will need to be reproduced and reported using a vanilla save game.
When reproducing a game bug in a vanilla save, make sure you are using the same mode _(singleplayer/co-op/dedicated multiplayer, survival/creative)_ as what you were using when the issue happened in the scenario.
To see a list of known game bugs or to report a new one please see: https://empyriononline.com/forums/bugs.24/

**Comment by mythras** [comment]
CV Sentinel and Colonist Bolters/Single Missile Turrets and small Modulated Laser turrets turned into HV/SV Med Modulated Lasers (UV)

**Comment by RavienGaming** [comment]
Edited original post to clarify that the devs are not aware of this issue as it has not been reported to them.

**Comment by shockr** [comment]
TLDR: misconfiguration of gameoptions.yaml causes defaults to apply, including DecayTime which causes the turbines to decay into hydroponics.

Ok, so I think I've figured it out.
The symptoms were: Windturbines were turning into fruit hydros, then into empty hydros.
At first I thought a weather change was causing it, then I started timing it and it always happened within 4 minutes. However it wasn't until I checked 'gopt' that I noticed all the config settings were defaulted. Including 'DecayTime'.
The reason for this was actually an error introduced into the gameoptions.yaml. Yaml is notorious for being very brittle. A tab instead of a space etc..
Replacing the gameoptions with one from a backup, the game settings took hold after rebooting the dedicated server.
With the DecayTime now at 0, the windturbines have remained unfruited.

**Comment by RavienGaming** [comment]
> TLDR: misconfiguration of gameoptions.yaml causes defaults to apply, including DecayTime which causes the turbines to decay into hydroponics.
> 
> Ok, so I think I've figured it out. The symptoms were: Windturbines were turning into fruit hydros, then into empty hydros. At first I thought a weather change was causing it, then I started timing it and it always happened within 4 minutes. However it wasn't until I checked 'gopt' that I noticed all the config settings were defaulted. Including 'DecayTime'. The reason for this was actually an error introduced into the gameoptions.yaml. Yaml is notorious for being very brittle. A tab instead of a space etc.. Replacing the gameoptions with one from a backup, the game settings took hold after rebooting the dedicated server. With the DecayTime now at 0, the windturbines have remained unfruited.


Very very unlikely this would be the cause. DecayTime, wind turbines, and crops have nothing to do with each other, nor would any game option affect any of those anyway. DecayTime is the time for "too small" structures to disappear.
Though with the game's coding anything is possible. Someone would need to report this to Eleon however as it's a game bug and outside of our control to do anything about.

Hope what you did actually fixed it, but I really have my doubts. This bug is simply random and will stop happening on its own. It wouldn't surprise me if it had something to do with the phases of the moon and the tides lol!

**Comment by shockr** [comment]
Fixed it. Repeated the bug. Maybe it's not DecayTime, but it most certainly is related to loading default config instead of the altered one and the most noticeable change that seems related is that setting. I'd have to individually tweak the config to see exactly which variable it is, but honestly I don't want to break our server anymore than we have already dealing with the fallout of the first time.

This only happens with mod blocks, so it could be related to how the game is handling blocks above the 2048 limit.

**Comment by 8836290** [comment]
Not sure if it is relevant but the same thing happens with some blocks when you copy vessels from old game saves to new ones by means of [Savegame Transfer Tool that copies bases/ships from one game to another](https://empyriononline.com/threads/tool-savegame-transfer-tool-copy-bases-ships-from-one-game-to-another.96142/page-10#post-467323)

**Comment by RavienGaming** [comment]
> Not sure if it is relevant but the same thing happens with some blocks when you copy vessels from old game saves to new ones by means of [Savegame Transfer Tool that copies bases/ships from one game to another](https://empyriononline.com/threads/tool-savegame-transfer-tool-copy-bases-ships-from-one-game-to-another.96142/page-10#post-467323)

That's actually expected behavior due to the way block IDs are saved to the individual save game, so when you copy a structure from one save to another the block mapping isn't the same.



> Fixed it. Repeated the bug. Maybe it's not DecayTime, but it most certainly is related to loading default config instead of the altered one and the most noticeable change that seems related is that setting. I'd have to individually tweak the config to see exactly which variable it is, but honestly I don't want to break our server anymore than we have already dealing with the fallout of the first time.
> 
> This only happens with mod blocks, so it could be related to how the game is handling blocks above the 2048 limit.

Would you be able and willing to report it to Eleon? They'll likely close it without even looking at it, but at this point yours is the closest lead I've seen.
https://empyriononline.com/forums/bugs.24/

**Comment by RavienGaming** [comment]
Note that this has been reported happening even on a fresh dedicated server save game that had yet to receive any game or scenario updates, so it's very unlikely to be due to any updates, though it does seem to happen more frequently around game or scenario updates.

There have been multiple claims of people "fixing" it or ideas of what the issue is, through various game option changes, config changes, only adding new blocks to the end of the block config list, manually fixing the block mapping config, to even the names of assets in the Unity bundles but so far nothing has been able to be consistently reproduced.

It doesn't help this seems to just randomly fix itself and stop doing it, so any supposed fix is impossible to tell whether it actually worked or was just luck until it's able to be reproduced at will.

This issue really does need to be reported to Eleon. The block mapping is controlled at the game level, not the scenario level, so scenarios have no control over the ID of any block that is over ID 2048. If the game is swapping blocks (especially at run time right in front of people's eyes as I've seen videos of) it's not something that a scenario could fix.
Unless someone knows of a cause within the scenario itself, this needs to be reported to the Empyrion developers using their bug report process with reproduction steps for it to have any hope of being fixed.

**Comment by 8836290** [comment]
> That's actually expected behavior due to the way block IDs are saved to the individual save game, so when you copy a structure from one save to another the block mapping isn't the same.

Is this a bug that needs to be reported too?

**Comment by RavienGaming** [comment]
> > That's actually expected behavior due to the way block IDs are saved to the individual save game, so when you copy a structure from one save to another the block mapping isn't the same.
> 
> Is this a bug that needs to be reported too?

No, because the block mapping data is saved in the save game and the game doesn't support moving structures between save games anyway.
You'd want to report that to the author(s) of those third party tools, though trying to mess around with the files in the save game folder will always be a risk.

**Comment by 8836290** [comment]
> You'd want to report that to the author(s) of those third party tools

I've done that years ago. Seems it can not be done by his side.

**Comment by RavienGaming** [comment]
Updated original post with the following:

If you have any information that can help track down the cause or how to prevent this issue from happening, please leave a reply here. It would be most appreciated.
Due to the bug report being marked as not a bug, we can only assume it's due to either a problem with server setup or management, or a problem in the scenario itself. At this time I have no idea what is causing it or how to prevent it from happening in the future so I'm asking the community for help. Please reply to this issue if you've found a way to reproduce and/or prevent this issue from occurring, or if you have any new information that can shed some light on why this happens.

Link to relevant bug report: https://empyriononline.com/threads/blocks-randomly-turning-into-other-blocks-mp.103070/

Video where this is seen happening in real time: https://youtu.be/doYz7VjTPjE?si=yLGjkPRiPqjrKmPt

**Comment by 8836290** [comment]
> Video where this is seen happening in real time: https://youtu.be/doYz7VjTPjE?si=yLGjkPRiPqjrKmPt

1.the video shows an orbit fight, is it really the video you wanted to give us?

2. One thing I did not mention is that I remember that the empty blocks issue was encountered mostly in RE1. In RE2 I have yet to notice it. I remind again that I only play SP games and I only copy vessels between old saves to new ones. The new ones I start every 6 months usually to update my games to newer versions of RE 2, though I have tested copying vessels created between RE2 close to each other, i.e. hot-fixes. 3.I am not sure 100% if copying a vessels from vanilla save game to RE does replace blocks or. I am 100% sure that copying RE vessels from RE save game to vanilla, blocks are missing but it makes sense since the missing blocks are not part of vanilla game. 

3.Replaced blocks between RE saves and when vanilla has been updated with new blocks result in different blocks replaced between RE saves. I know that for sure since I mostly use the same RE version of  a bunch of my favorite vessels. Example vanilla version a.a and RE version AA I copy from RE a vessel and lets say the generators are replaced with i.e Imperfect Pentaxid Seed Crystal AND vanilla is updated to version a.b and gets new blocks AND RE gets updated to version A.B so  I copy vessels form  old to new RE saves the generators are replaced with something else. The PITA was to remove the empty block in order to place the proper RE block, example food processor for SV/HV.  I remember this for sure for RE1 it does not happen to RE2.

4.Extra info : The last year or so I tend to copy vanilla vessels to RE2 because I like to test vanilla updates. Both experimental OR public versions. Meaning sometimes one or the other. When I return to RE2 I always copy those vessels and they do not miss blocks but of course similar type ones are replaced, example the thrusters of SVs are replaced with the lower tier (Ion version) of RE.

PS This bug is a nice puzzle. My knowledge with Data Bases is poor for now, but  this and the knowledge of this game as a player and amateur moder, are enough to get a hunch that it might be a DB issue. I am learning about DBs myself attending courses I am starting to inspect Empyrion's for academic purposes and I hope in the near future with more knowledge to provide more help to you or the devs. With proper vernacular. Of course we are not sure if it is a DB bug or how it is parsed/used by the game in terms of code.

**Comment by RavienGaming** [comment]
Sorry 8836290, but this has nothing to do with copying structures from one save to another.
Like I had mentioned, when you copy structures between saves, the block mapping IDs can be different, resulting in the issues you experience. That is 100% expected behavior and not relevant to the problem _this_ issue is about, which is blocks randomly turning into other blocks within an existing save in real time (as seen in the video).

**Comment by RavienGaming** [comment]
Please reply to this issue if you've encountered this problem **AFTER** the build 43 update to the scenario.

**Comment by Ashenator-01** [comment]
This wont help you directly I am afraid... However I have 24 hour hydro bays that are using the same method as your hydro bays, and I copied the same settings for the build 43 update. I just got a report of someone picking up one of my pentaxid bays and it turning into a grain bay.

Also what I have noticed on this issue is the addition of blocks in the BlockConfig.ecf file when used like : 
{ +Block Name: CargoDrumLarge, Ref: ContainerMS01 (As added in build 43)
I am under the impression the +Block causes servers to dynamically adjust the block ID#s upon loading and simply removing the + seems to force a more stable loading process. Maybe Taylen can share some light into how the { +Block Name:/{ Block Name: system truly works behind the scenes. However so far this is just my theory...

I forgot to add, the last time I had a block change before this morning was when Mobile Furnaces were added. I did a restore on my server when this occurred and removed the plus from the +Block of the furnace and the issue did not re-occur.

**Comment by 8836290** [comment]
I quote from EGS site Taelyn's view of the issue : 

"All I can say is that if you add new blocks ALWAYS add them complete at the bottom, and never between
Do NOT delete anything, set it to hidden if you dont want it to show up. This are the 2 rules we follow internally and we never had an issue, not in SP or MP"

For me, it is clear that : 
1) Those two rules are not followed hence the mess. 
2) developers knows best. Obviously. They can look at code we can not. 
3) I wonder why this bug does not appear to other scenarios. I play a bunch of them from time to time and NEVER encountered the bug while copying vessels between old version -> new version saves.
4) The fact that SP and block replacement AND block replacement on MP games happens must be the same. And as far as I know it happens on YOUR scenario only. 

Let me give an I idea that might help. Spare some of your time and do a test 

a) Take the vanilla configuration and replace a block between the vanilla block range and add a couple of blocks at the end of it. 
b) Do the same but this time FOLLOW the rules Taelyn told you. 

It would be like you start again creating two different versions of a new scenario from scratch. Then publish it so we can test it. I will happily test both SP  for the copy bug (which for some reason you think it has nothing to do with this bug stated here and on EGS site, but all facts point against that), and I will also do tests on the MP game.  Just my two cents.

**Comment by RavienGaming** [comment]
8836290, please stop. The issues of copying structures between save games has nothing to do with this issue and you're muddying things by trying to say that they do. I appreciate that you're trying to help but the RE2 GitHub is not really the place to discuss copying structures between save games like that.

**Comment by RavienGaming** [comment]
****
> Also what I have noticed on this issue is the addition of blocks in the BlockConfig.ecf file when used like : { +Block Name: CargoDrumLarge, Ref: ContainerMS01 (As added in build 43) I am under the impression the +Block causes servers to dynamically adjust the block ID#s upon loading and simply removing the + seems to force a more stable loading process. Maybe Taylen can share some light into how the { +Block Name:/{ Block Name: system truly works behind the scenes. However so far this is just my theory...
> 
> I forgot to add, the last time I had a block change before this morning was when Mobile Furnaces were added. I did a restore on my server when this occurred and removed the plus from the +Block of the furnace and the issue did not re-occur.

Interesting, we'll try that. The + was used to make the block accessible in the old config.ecf file which is no longer used anyway. I don't know what else that + does. We'll try removing it and seeing if that makes a difference.

**Comment by RavienGaming** [comment]
Reforged Eden 2 Beta has been updated to Build 44:

Server admins, this makes a change to the block config file to remove the + in Block Id and Block Name in the blocksconfig.ecf file in an attempt to fix the block mapping bug. Please report any issues of blocks turning into other blocks after updating to build 44 on here.

**Comment by RedScourge** [comment]
I used to have this issue in RE1. I completely avoided it ever since by maintaining my own version of the scenario where I add all new blocks to the end of the BlocksConfig file rather than mixing them in next to similar blocks (since the blocks do not have a hard-coded ID, I suspect this causes problems whenever the server randomly decides to reload IDs from the scenario config rather than consulting the blocksmap file)

**Comment by RavienGaming** [comment]
Any instance of this issue happening needs to be reported to Eleon with the full logs, a copy of the save game before and after the block change, a copy of the affected blueprint(s) before and after the block change, and a copy of the scenario before and after the block change.

**Comment by 8836290** [comment]
> Any instance of this issue happening needs to be reported to Eleon with the full logs, a copy of the save game before and after the block change, a copy of the affected blueprint(s) before and after the block change, and a copy of the scenario before and after the block change.

Do we have to issue a new bug report on EGS bug forums, or is there an open one tracking it? If it is the second provide the link here so every one go there and upload the files you requested.

**Comment by RavienGaming** [comment]
It's linked in the first post here.

**Comment by RedScourge** [comment]
Eleon famously ignores all bug reports involving custom scenarios. An additional problem is it is not easy to track the issue in the logs, as it seems to appear only after a backup/restart and if a server has no players for a day or two, you may have to check through days of logs, and then sift through thousands of garbage lines if you run the ASTIC mods, etc. Because no one wants to run a vanilla scenario, which is probably required to track down issues like this, I think we're just screwed, and destined to have this happen now and then. As I said, ever since I started adding all new blocks to the end of the BlocksConfig file, I have not had this issue, whereas I have had this issue on at least 3 separate occasions before (usually months into a playthrough).  I suspect it is so much more likely to happen to me because I update my server to install every RE update and wipe maybe every 6-12 months, whereas many others do not update as often and wipe more often.

My working theory is that occasionally, the server decides to reject the blocksmap file and rebuilds it from config, and all the blocks with no ID numbers, which is most of the RE devices like warp drives, weapons, hydroponics, etc, get corrupted, because the server interprets the structure .EPB files with the old blocksmap through the eyes of a new blocksmap, but because the config has had new items mixed in part way through with no ID numbers, those new items take the place of the items that come after, and then you get things like warp drives replaced with posters and new effect projector blocks, etc. If so, this is a issue that is destined to be almost exclusive to custom scenarios, which means a high chance that Eleon never bothers to look into it, even if they change their ways and start caring about fixing bugs instead of their obsession with updating the vanilla scenario.

**Comment by RavienGaming** [comment]
@Ashenator-01 @shockr were you using a RAM drive for your servers by chance?

**Comment by shockr** [comment]
Unfortunately not. The dedicated server was just running with no special configuration on a Windows 11 machine as a standard application.

The only common denominator I was able to find was that the game options were being set to defaults because the existing gameoptions.yaml became invalid.

**Comment by B-iggy** [comment]
> Eleon famously ignores all bug reports involving custom scenarios
Bugs reported to me are never ignored. In rare cases some just can't be fixed unfortunately.

This bug did not happen on my server, so we have to find out what the others are doing different.
We solved the Spanj case at least.

**Comment by B-iggy** [comment]
@shockr what does invalid mean?

Windows 11 machine? A dedicated server not running Windows Server as operating system?

Do you use any symlinks for your scenario or savegame?

**Comment by ACEighty7** [comment]
@B-iggy What was Spanj's issue? how was it solved? I cannot see here

**Comment by B-iggy** [comment]
The issue was that his infrastructure is based on symlinks / some RAM Drive linking. That way he got this error in his playfield log:
`-ERR- Error on loading file E:\BrokenEden/Saves/Games/BE2-B38/blocksmap.dat: Sharing violation on path E:\BrokenEden\Saves\Games\BE2-B38\blocksmap.dat`

which is the reason why blocks are morphing, since this file is responsible for it.

Not sure myself exactly how/what is going on in the background but kinda wild. I can only tell from my own experience, working with symlinks is a straight way to issues for Empyrion.

The knowledge about it, can be seen as solved. It is up to him now, to do try and error sessions to make his infrastructure work or do it the normal way.

**Comment by ACEighty7** [comment]
Oh, completely different to me then. I dont uses a RAMDisk due to the volatile nature of them. Power outage means all data lost unless regularly backed up to a proper drive.  I get these errors too. what is Symlinks? Not sure if i use it. My server is very basic. I just use EAH

**Comment by B-iggy** [comment]
Symlinks for computers are references to a file or a folder.
For example to not have 2 same folders at location X and Y but instead pointing location X to Y.

Probably you do not use them then.

I do not see your specific usecase in this ticket here. What server you are running, what happened, etc.
Only thing I can suggest blindly to you is a global / folder search through your whole Logs folder (`steamapps\common\Empyrion - Dedicated Server\Logs`) for the term `blocksmap.dat` and report back if you found something.

**Comment by ACEighty7** [comment]
I own and run my own machine. v1.15 with RE2 build 44. Same errors as Spanj. Only mine seem to occur when the POI is spawning in. Wiping the playfield entirely fixed it for me before so it could pull a fresh prefab. But even that isn't working now. As soon as the structure spawns I get an in game error and this same blocksmap.dat error in the logs. 

So I guess we are no closer to solving it. Maybe it's just a hardware issue. Not responsive enough.

**Comment by B-iggy** [comment]
> So I guess we are no closer to solving it

Knowing the enemy is always a step closer than wandering in the dark.
You have to figure out now, what is causing a sharing violation on your machine.
Figure out any tiny possibility that could cause a read/write access to that file during runtime of the server (e.g. backups, mods, scripts, Windows Event manager, etc.)

Again, I do not have this issue and for the record I'm hosting the biggest MP servers since 10 years.
On Windows Server 2022 Standard as operating system with only Process Lasso as external software handling the load (I would say this program is mandatory for anyone trying to offer a somewhat serious MP environment for their players. Short: without you have issues)

**Comment by ACEighty7** [comment]
Now that you mention it, this never happened on my older Windows 10 server RIP. I'm currently running a basic, unbloated Windows 11 but updates and background tasks will obviously still happen. It's only a small server for my Discord and Twitch community with a maximum capacity of 28 playfields (RAM Limitations) Run entirely via EAH. 

My main go to MP server is with the ANVIL community, I just built my own for those of us that are a little slow for the game seasons. So as you can probably guess from that statement, I am a complete novice.

Project Lasso is something I will need to check out. But I will also try running the server from another SSD. I'll even check if there is a new BIOS or drivers out lol

**Comment by RavienGaming** [comment]
And please feel free to report your findings on the bug report thread here: https://empyriononline.com/threads/blocks-randomly-turning-into-other-blocks-mp.103070/

So far this seems like the best lead we have so any additional information anyone has to post there could be helpful.

**Comment by shockr** [comment]
Yes, this was running on a spare Windows 11 gaming machine. The problems only arose when the yaml syntax was incorrect so the game discarded the file and used default settings.

When this happened, wind turbines were spontaneously turning into fruit hyrdobays. After figuring out that I had an error in the yaml I fixed the file and the problem stopped happening.

**Comment by ACEighty7** [comment]
Separate SSD seems to be working for me though still early days. Like the video I sent to Ravien I tried to recreate the issue by Discovering, spawning, leaving, wiping and returning. Without the separate SSD It would do it every time. With the separate SSD I wasn't able to repeat it. But it is still early days. Maybe it makes itself known after a few EAH Timetabled POI wipes. I did also update my machine's BIOS as it was missing the fix for SSDs and fTPM modules being broken by a Windows Update a while ago.

I do not use steamcmd or any scripts for updating. The server is always updated via steam and backed up (Folder copy) while shut down, the scenario folder is updated via drag 'n' drop of the files. I went through my Windows Event viewer and Microsoft Defender was running at the same time as the Playfield logs flagged the Blocksmap.dat error

Nothing automated, nothing fancy. Thought I should mention that. I will keep you updated.

**Comment by Ashenator-01** [comment]
No my server does not run under a RAM drive and does not use symlinks. I'm running up to 7 instances of Empyrion via EAH and about 25-30 other game servers on my personal rig.

My rig : Dell T630 running Windows Server 2022 Datacenter with 28 cores 56 threads with 256 GB RAM.

I've only had block changes after updates, but in my case I run a custom scenario with a lot more added blocks then Vanilla RE2. I have not had another issue since the (+) in +block was removed.

Edit : I'm also using 10x 480 GB 2.5 in SSDs in a raid 10 array as the only virtual drive.

**Comment by ACEighty7** [comment]
Update:

Since the BIOS update and running the server form a dedicated NVME I have had no blocksmap read errors or any spawn in issues. Server has been regenerating POIs, Deposits and is responding flawlessly. 

I also partitioned the NVME drive that the server is installed on and loaded up the Vanilla server up to run at the same time, hoping to simulate any potential interruption., I am running identical EAH Timetable tasks across both servers at the same times.

My OS installation hasn't performed an update recently so I cannot rule that out yet. Still keeping an eye on the server logs and windows event viewer. I will keep on testing and keep you posted

**Comment by ACEighty7** [comment]
Ok, I was able to recreate it yet again and came across these perflib 1008 errors in windows event viewer at the time of the affected POI spawn in and also at the time of the Blocksmap error in the playfield logs (UCH Vessel with SV Thrusters sticking out of the top)

Maybe worth taking a look? I will continue to keep an eye out. Thinking it must be a Windows background task.

<img width="1219" height="498" alt="Image" src="https://github.com/user-attachments/assets/7f971f96-ec67-46b6-9f4e-3a704e1893fb" />

**Comment by RavienGaming** [comment]
> Ok, I was able to recreate it yet again and came across these perflib 1008 errors in windows event viewer at the time of the affected POI spawn in and also at the time of the Blocksmap error in the playfield logs (UCH Vessel with SV Thrusters sticking out of the top)
> 
> Maybe worth taking a look? I will continue to keep an eye out. Thinking it must be a Windows background task.
> 
> <img alt="Image" width="1219" height="498" src="https://private-user-images.githubusercontent.com/268750356/569772726-7f971f96-ec67-46b6-9f4e-3a704e1893fb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzQ1Njg3NTEsIm5iZiI6MTc3NDU2ODQ1MSwicGF0aCI6Ii8yNjg3NTAzNTYvNTY5NzcyNzI2LTdmOTcxZjk2LWVjNjctNDZiNi05ZjRlLTNhNzA0ZTE4OTNmYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMyNlQyMzQwNTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MmY5ZGYzMWI4NTE3MmJhY2I2NjQ1YTI2MWRmNzk1YTU0YTg0MzZlODA1MDAzOGJmYmY5Yzg1OTEyZTM1ZDM0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.xTiFYiKAL1zyJS-l6TAo8i9j8hEm0CpPKh9EuRWBIfA">

This is in the current build of the scenario?
Hmm... You'd really want to post any info you have on the bug report on Eleon's forums. There's nothing else I can think of to change in the scenario outside of what I've already done the last 2 updates.

**Comment by ACEighty7** [comment]
Yeah, this is with the current scenario. I am starting to think that this is nothing to do with the game or your scenario. As far as I can see these read interrupts are happening when my OS is busy doing something else in the background. If EAH is having issues reading the blocksmap file I am wondering if it is just selecting an ID sequentially. I am still investigating. I have turned off auto updates, will try that first, then I will disable windows defender and assign a manual schedule, see if it is that. If it's just my setup and hardware, this does note bode well for hosted/rented servers

**Comment by RavienGaming** [comment]
> Yeah, this is with the current scenario. I am starting to think that this is nothing to do with the game or your scenario. As far as I can see these read interrupts are happening when my OS is busy doing something else in the background. If EAH is having issues reading the blocksmap file I am wondering if it is just selecting an ID sequentially. I am still investigating. I have turned off auto updates, will try that first, then I will disable windows defender and assign a manual schedule, see if it is that. If it's just my setup and hardware, this does note bode well for hosted/rented servers

Yes please post your findings on the bug report that's linked earlier.

**Comment by shockr** [comment]
Just came to mention the latest game update seems to have effected wind turbines again. This time no change with the game options yaml, and all the settings in gopt seem fine. Stumped.

**Comment by RavienGaming** [comment]
> Just came to mention the latest game update seems to have effected wind turbines again. This time no change with the game options yaml, and all the settings in gopt seem fine. Stumped.

Are you also running build 44 of the scenario?

**Comment by shockr** [comment]
Yes, however there had been no problem until yesterday's (26th) updates for the game itself.

**Comment by RavienGaming** [comment]
> Yes, however there had been no problem until yesterday's (26th) updates for the game itself.

Like mentioned earlier, there's nothing more I can do to assist with this issue unfortunately as it seems to be beyond the scenario.
Any information about this would need to be posted to that bug report thread and collaborated with other server owners to find the root cause.

**Comment by ACEighty7** [comment]
> > Yes, however there had been no problem until yesterday's (26th) updates for the game itself.
> 
> Like mentioned earlier, there's nothing more I can do to assist with this issue unfortunately as it seems to be beyond the scenario. Any information about this would need to be posted to that bug report thread and collaborated with other server owners to find the root cause.

Beginning to think the same here. I honestly believe it's not the game itself but something interrupting. the only thing I can think of is somehow get the server to make more attempts at loading that file or increase the detection window. Hardware and software has changed a lot over the past 10 years, Windows 11 especially is an absolute nightmare when it comes to performance management. Parking cores, shutting down drives (sometimes bricking them,) intrusive virus scans and auto updates too. 

Actually considering switching to Server and use a process manager like Rexxus uses. But first I will check them disabled one by one with my current setup

**Comment by RavienGaming** [comment]
Marking this "fixed" because while the underlying issue is most definitely not fixed, going forward we will only add new blocks to the very end of the block config file which means for new save games when the issue does occur block IDs won't change.
This unfortunately won't help any save games started before the current build 44 of RE2, but any saves started in build 44 or later should not suffer block ID changes when the block mapping file can't be read.

Also keeping the "game bug" label until or unless the game devs fix the actual problem, but that is unlikely.

I will keep this issue open for a few months to give existing servers a chance to finish up their current seasons, in case this issue happens again for them so it's easier to find this issue.

**Comment by ACEighty7** [comment]
Ok I will give my guys a heads up. Either continue to play and I fix by resetting the playfield and restarting the server or start a new save on 44.
